proposals/269/transport.md

```diff
@@ -0,0 +1,92 @@
+
+## Proposed interface
+
+In pseudocode:
+```
+interface Transport {
+  send(LineageEvent event);
+}
+```
+
+The send method should be synchronous and return when underlying system confirms 
+that message has been received. This can be for example HTTP response in 200 range, 
+successful fsync on file descriptor, or successful Apache Kafka flush.
+
+Potential error handling is deliberately unspecified on overall interface level, 
+but should be specified and consistent for particular language implementation. 
+The reason for that is difference between idiomatic error handling between different 
+languages: while Python and Java prefer exceptions, Go prefers multiple return values, 
+and Rust prefers monadic `Result<T,E>`. 
+
+However, `Clients` and `Transports` utilizing exceptions should be careful to not directly expose 
+integrations to exceptions thrown by underlying libraries - integrations should be 
+independent of `Transports`. 
+
+
+`Transports` are implemented for particular language's `Client`. There is no guarantee 
+of availability of any `Transport` on instance that implementation exist for different 
+language, with following exception.
+Any reference implementation of `OpenLineage` `Client` should include default 
+`HTTP Transport`. 
+
+## Instantiation
+
+Integration code instantiation is usually controlled by system for which they are 
+written - for example Airflow spawns `LineageBackend` instance. Then, integration
+code would spawn `Client`, which would be responsible for spawning relevant 
+`Transport`. This is contrary to usual dependency injection approach, which creates an issue:
+that `Client` needs to be aware of which `Transport` to spawn.
+
+This would be solved by standardizing `TransportFactory` interface, which constructs
+`Transports`.
+
+```
+interface TransportFactory {
+    Transport create(Config config);
+}
+```
+
+Transports can define additional configuration variables, that would be provided 
+by `TransportFactory` in form of `Config` key-value map.
+Same `Transport` implementations for different languages should
+be careful to keep configuration consistent, however, some differences are accepted
+if it improves user experience.
+
+`Clients` provide default implementation of `TransportFactory` which creates `Transports` 
+based on `OPENLINEAGE_TRANSPORT` environment variable.
+Buildin `Transports` are specified via case-insensitive aliases. 
+`OPENLINEAGE_TRANSPORT=http` sets transport to `HTTP`.
+
+Default `TransportFactory` can load non-buildin transports based on fully-qualified class names. 
+The fully-qualified name can differ between languages. For example, Java implementation
+of `Apache Kafka Transport` can be `io.openlineage.transport.KafkaTransport`.
+
+The default implementation also passes all environment
+variables fulfilling `OPENLINEAGE_TRANSPORT_*` as `Config`.
+Different `TransportFactories` could utilize other configuration methods, for example
+YAML files. 
```

Comment:
Personally, I'd vote for the yml option being the default, as in @wslulciuc's comment [here](https://github.com/OpenLineage/OpenLineage/issues/269#issuecomment-1005273574)

Reply:
I agree, but I think we can provide (initially) 3 options for configuring the transport:

1. An `openlineage.yml` at the root of the classpath
2. An `openlineage.yml` file in a `.openlineage/` folder in the user's home directory (~/.openlineage/openlineage.yml
3. Environment variables